### PR TITLE
Move all HTTP headers into AcceptHeaders helper class

### PR DESCRIPTION
**Pull request by ryangribble**
As mentioned in #996 this is a follow up PR that tidies up all hardcoded HTTP headers used within Octokit.net, into the AcceptHeaders helper class that was added

- Replaced all uses of hardcoded HTTP header strings to declare/use entries in the AcceptHeaders helper class, like the protected branches API header in #996 does

- Left test methods alone (ie still hardcoding their expected accept strings) since we shouldn't share definitions between test and implementation code, probably shouldnt even share it between 1 test and another

This change was fairly precise/surgical in nature and all unit and integration tests are passing, so hopefully it's all good!

### CODE REVIEW

File: Octokit/Helpers/AcceptHeaders.cs [hunk 1]
```diff
@@ -2,6 +2,14 @@
 {
     public static class AcceptHeaders
     {
+        public const string DefaultJson = "application/vnd.github.quicksilver-preview+json; charset=utf-8, application/vnd.github.v3+json; charset=utf-8";
+
+        public const string DefaultHtml = "application/vnd.github.html";
+
+        public const string ReleaseApi = "application/vnd.github.v3";
```
Comment (shiftkey) [inline Octokit/Helpers/AcceptHeaders.cs]: Let's name this something like `StableVersion` as its the replacement for the releases API now that's out of preview

File: Octokit/Helpers/AcceptHeaders.cs [hunk 2]
```diff
@@ -2,6 +2,14 @@
 {
     public static class AcceptHeaders
     {
+        public const string DefaultJson = "application/vnd.github.quicksilver-preview+json; charset=utf-8, application/vnd.github.v3+json; charset=utf-8";
```
Comment (shiftkey) [inline Octokit/Helpers/AcceptHeaders.cs]: This is related to [repository redirects](https://developer.github.com/changes/2015-04-17-preview-repository-redirects/) and then the latest stable version - could we rename it to something like `RedirectsPreviewThenStableVersionJson` ?

File: Octokit/Helpers/AcceptHeaders.cs (hunk 1 above)
Comment (shiftkey) [inline Octokit/Helpers/AcceptHeaders.cs]: It can probably be cleaned up so that the defaults are loaded in...

**Comment by shiftkey** [comment]
Looking good, just a few little suggestions on Naming Things™

**Comment by ryangribble** [comment]
No worries, naming things is the hardest part so always keen for feedback :stuck_out_tongue:

**Comment by Haacked** [comment]
I like where this is headed! I agree with the naming suggestions @shiftkey proposed.

**Comment by ryangribble** [comment]
Cool, I'll push some changes later this week when I'm back (currently overseas)

**Comment by ryangribble** [comment]
Hey guys, Im back from holidays and have pushed up some renames as well as merged latest master into my branch.  Let me know what you think of the names

**Comment by shiftkey** [comment]
@ryangribble looks good to me, thanks for the work!

**Pull request merged**